Add test for istio.io task: Secure Ingress Gateway with SDS

JimmyCYJ opened the pull request:
Please provide a description for what this PR is for.
Automate task https://preliminary.istio.io/docs/tasks/traffic-management/ingress/secure-ingress-sds/, verified with kind
Tested via
go test ./... -p 1 --istio.test.env kube --istio.test.kube.config ~/.kube/kind-config-kind --log_output_level=tf:debug --istio.test.hub=<hub> --istio.test.tag=<tag> -timeout 30m --istio.test.nocleanup=false --istio.test.kube.minikube -v

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

JimmyCYJ commented:
I am thinking about passing parameter to newStepScript, so that we can run script with parameters. That will reduce the number of scripts. The current version strictly follows the task and adds one script for each step.

istio-testing commented:
@JimmyCYJ: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-racetest-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3301) | `/test istio-racetest-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-shellcheck-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2282) | `/test istio-shellcheck-master`
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-racetest-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3301) | `/test istio-racetest-master`
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

rlenglet commented on pkg/test/istio.io/examples/example.go:
```diff
@@ -118,6 +118,7 @@ func (example *Example) Run() {
 			}
 			for _, step := range example.steps {
 				output, err := step.Run(kubeEnv, example.t)
+				example.t.Log(output)
```
Duplicate with the same code below. Remove one of the `Log` calls. I would keep yours and remove the one below.

brian-avery replied:
Agree. And the call to example.t.Fatal(output) should be example.t.Fatal(err)

JimmyCYJ replied:
This ```Log``` call is kept, and ```Fatal(output)``` is changed to ```Fatal(err)```. Thanks.

rlenglet commented on pkg/test/istio.io/tasks/security/secure-gateway-sds/secure_gateway_sds_test.go:
```diff
@@ -0,0 +1,131 @@
+func TestMain(m *testing.M) {
+	// Integration test for the ingress SDS Gateway flow.
+	framework.
+		NewSuite("sds_ingress_gateway_test", m).
```
We want the test names to match the doc names.
The URL is `auth-sds`, so use the same name here. And rename this test's directory into `auth-sds`, and this file into `auth-sds_test.go`.

JimmyCYJ replied:
All the directory name, file name and test name are updated. Thanks.

rlenglet commented on pkg/test/istio.io/tasks/security/secure-gateway-sds/httpbin-deployment.sh:
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
```
I can't find these commands in the https://preliminary.istio.io/docs/tasks/security/auth-sds/ page.
All the scripts in these tests are supposed to be literal copy-and-paste from the docs, to check that the instructions we give to users in the docs are actually working.
Rewriting those instructions defeats the purpose of writing these tests.

rlenglet commented on pkg/test/istio.io/tasks/security/secure-gateway-sds/mtls-go-example.sh:
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Same here, and in your other scripts.
Please replace all your hand-written scripts with copy-and-paste from the doc's shell snippets.

JimmyCYJ replied:
Sorry I put wrong link in the PR description. The correct link is https://preliminary.istio.io/docs/tasks/traffic-management/ingress/secure-ingress-sds/.
These scripts are copy-and-paste from doc's shell snippets.

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-racetest-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3301) | `/test istio-racetest-master`
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-shellcheck-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2505) | `/test istio-shellcheck-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-lint-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-lint-master/2743) | `/test istio-lint-master`
istio-unit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5223) | `/test istio-unit-tests-master`
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-racetest-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3301) | `/test istio-racetest-master`
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-shellcheck-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2505) | `/test istio-shellcheck-master`
integ-new-install-k8s-presubmit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1890) | `/test integ-new-install-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-racetest-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3301) | `/test istio-racetest-master`
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-shellcheck-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2505) | `/test istio-shellcheck-master`
integ-new-install-k8s-presubmit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1890) | `/test integ-new-install-k8s-presubmit-tests-master`
istio-unit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5502) | `/test istio-unit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-mixer-k8s-presubmit-tests-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-mixer-k8s-presubmit-tests-master/3887) | `/test integ-mixer-k8s-presubmit-tests-master`
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-shellcheck-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2505) | `/test istio-shellcheck-master`
integ-new-install-k8s-presubmit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1890) | `/test integ-new-install-k8s-presubmit-tests-master`
istio-unit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5502) | `/test istio-unit-tests-master`
istio-racetest-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3594) | `/test istio-racetest-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
integ-new-install-k8s-presubmit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1890) | `/test integ-new-install-k8s-presubmit-tests-master`
istio-unit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5502) | `/test istio-unit-tests-master`
istio-racetest-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3594) | `/test istio-racetest-master`
istio-shellcheck-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2510) | `/test istio-shellcheck-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-unit-tests-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-unit-tests-master/5502) | `/test istio-unit-tests-master`
istio-racetest-master | 87da7d5fc9da9a0a5dfd0dea8d312f14c28866c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-racetest-master/3594) | `/test istio-racetest-master`
istio-shellcheck-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2510) | `/test istio-shellcheck-master`
integ-new-install-k8s-presubmit-tests-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1895) | `/test integ-new-install-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
istio-pilot-multicluster-e2e-master | 61adcf02ae726cadc5169b3b77e88124ca7753e9 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-pilot-multicluster-e2e-master/3996) | `/test istio-pilot-multicluster-e2e-master`
istio-shellcheck-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/istio-shellcheck-master/2510) | `/test istio-shellcheck-master`
integ-new-install-k8s-presubmit-tests-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1895) | `/test integ-new-install-k8s-presubmit-tests-master`
integ-security-k8s-presubmit-tests-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-security-k8s-presubmit-tests-master/4573) | `/test integ-security-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-presubmit-tests-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1895) | `/test integ-new-install-k8s-presubmit-tests-master`
integ-security-k8s-presubmit-tests-master | 018bbfb0a816191677c6749b548b6e2f0aacc025 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-security-k8s-presubmit-tests-master/4573) | `/test integ-security-k8s-presubmit-tests-master`
e2e-dashboard-master | 385f557e9a25f3dc27a215223eceae31df12c56d | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/e2e-dashboard-master/5469) | `/test e2e-dashboard-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@JimmyCYJ: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
e2e-dashboard-master | 385f557e9a25f3dc27a215223eceae31df12c56d | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/e2e-dashboard-master/5469) | `/test e2e-dashboard-master`
integ-security-k8s-presubmit-tests-master | 385f557e9a25f3dc27a215223eceae31df12c56d | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-security-k8s-presubmit-tests-master/4598) | `/test integ-security-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

JimmyCYJ commented:
/retest

istio-testing commented:
@JimmyCYJ: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-security-k8s-presubmit-tests-master | 385f557e9a25f3dc27a215223eceae31df12c56d | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-security-k8s-presubmit-tests-master/4602) | `/test integ-security-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

JimmyCYJ commented:
/retest

JimmyCYJ commented:
@rlenglet Please take a look. Thanks.

rlenglet commented:
OK, it seems you're testing https://preliminary.istio.io/docs/tasks/traffic-management/ingress/secure-ingress-sds/, as indicated in the test comments. You need to rename the directory to match that (`security` -> `traffic-management/ingress`).

rlenglet commented on pkg/test/istio.io/tasks/security/secure-ingress-sds/secure-ingress-sds_test.go:
```diff
@@ -0,0 +1,136 @@
+	if cfg == nil {
+		return
+	}
+	// Enable SDS key/certificate provisioning for ingress gateway.
```
To match the documentation, I'd explicitly add this:
```
cfg.Values["gateways.istio-egressgateway.enabled"] = "false
```

JimmyCYJ replied:
Done. Thanks.

rlenglet commented on pkg/test/istio.io/tasks/security/secure-ingress-sds/curl-helloworld-tls-gateway-gke.sh:
```diff
@@ -0,0 +1,19 @@
+echo "SECURE_INGRESS_PORT ${SECURE_INGRESS_PORT}"
+export SECURE_INGRESS_PORT
+
+echo "curl -v -HHost:helloworld-v1.example.com \
```
Add `set -x` at the top of the script. It will have the same effect and will be simpler to maintain.

JimmyCYJ replied:
Thanks for the advice. Done.

istio-testing commented:
@JimmyCYJ: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-presubmit-tests-master | 5537f16d9d63fa4c6e4cdd3343702dfcb97a30be | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1972) | `/test integ-new-install-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

JimmyCYJ commented:
> Overall, this is really good, thanks!
> A couple minor things to fix before merging:
> 
> * Rename the test's directory to match the directory in istio.io.
> * Replace the explicit `echo` commands in the scripts with `set -x`, to make it more readable.

Thanks for the advice. I have made changes accordingly. Please take a look.

rlenglet commented:
@JimmyCYJ Small pro-tip 🙂: to reassign this PR to me for review, expand the `x changes requested` section, click on the `...` icon on the right of the line saying `rlenglet requested changes`, then click `Dismiss review`. Then it will show up again as waiting for my review.

rlenglet commented on pkg/test/istio.io/tasks/traffic-management/ingress/secure-ingress-sds/cleanup.sh:
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [[ -d "mtls-go-example" ]]; then
```
Remove the `if`s. `rm -rf` will work regardless.

JimmyCYJ replied:
done. thanks.

istio-testing commented:
@JimmyCYJ: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-presubmit-tests-master | b44cde83207bf9351d4f111306244caea8b456a1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/1996) | `/test integ-new-install-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

JimmyCYJ commented:
@nmittler Could you please take a look?

ozevren commented:
The placement of the tests are not right. These should go into `istio/tests/...`, not `pkg/test/...`.
It should also have the right owners set at that folder.

istio-testing commented:
@JimmyCYJ: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

JimmyCYJ commented:
> The placement of the tests are not right. These should go into `istio/tests/...`, not `pkg/test/...`.
> It should also have the right owners set at that folder.

I have moved all tests into /tests/istio.io/, and added code owners. Please take a look. Thanks.

istio-testing commented:
@JimmyCYJ: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-new-install-k8s-presubmit-tests-master | 87b8e72d534608563f4e30b24802b7c23c3feef4 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/16154/integ-new-install-k8s-presubmit-tests-master/2038) | `/test integ-new-install-k8s-presubmit-tests-master`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->
